### PR TITLE
Add view command to find specific student via name or id

### LINKED ISSUE #5: As a Tutor I can view Single students detail
...so that I can see the individual detail for a single student

### PR DESCRIPTION
**Pull request by 0nesheep**
closes #5 

Adds functionality of view -name {NAME} or view -id {ID} to search for students

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/ViewCommand.java [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+ */
+public class ViewCommand extends Command {
+
+    public static final String COMMAND_WORD = "view";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Views the specific student using their ID or name\n"
+            + "Parameters: NAME or ID\n"
+            + "Example: " + COMMAND_WORD + " John OR " + COMMAND_WORD + " 12345";
+
+    private final NameContainsKeywordsPredicate namePredicate;
```
Comment (KinTatHo) [inline src/main/java/seedu/address/logic/commands/ViewCommand.java:23]: SCREAMING_SNAKE_CASE for final variables

File: src/main/java/seedu/address/logic/commands/ViewCommand.java [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    private final IsSameIdPredicate idPredicate;
```
Comment (KinTatHo) [inline src/main/java/seedu/address/logic/commands/ViewCommand.java:24]: this too

File: src/main/java/seedu/address/logic/parser/AddressBookParser.java [hunk 3]
```diff
@@ -8,15 +8,7 @@
...
-import seedu.address.logic.commands.AddCommand;
-import seedu.address.logic.commands.ClearCommand;
-import seedu.address.logic.commands.Command;
-import seedu.address.logic.commands.DeleteCommand;
-import seedu.address.logic.commands.EditCommand;
-import seedu.address.logic.commands.ExitCommand;
-import seedu.address.logic.commands.FindCommand;
-import seedu.address.logic.commands.HelpCommand;
-import seedu.address.logic.commands.ListCommand;
+import seedu.address.logic.commands.*;
```
Comment (KinTatHo) [inline src/main/java/seedu/address/logic/parser/AddressBookParser.java:11]: i think this is not recommended, should be imported one by one

File: src/main/java/seedu/address/model/person/Id.java [hunk 4]
```diff
@@ -0,0 +1,36 @@
+package seedu.address.model.person;
+
+public class Id {
+    public static final String MESSAGE_CONSTRAINTS =
+            "Ids should only contain numbers";
+    public final String id;
```
Comment (KinTatHo) [inline src/main/java/seedu/address/model/person/Id.java:6]: SCREAMING_SNAKE_CASE

File: src/main/java/seedu/address/model/person/Id.java [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+
+    public Id(String id) {
+        this.id = id;
+    }
+
+    public static boolean isValidId(String test) {
+        try {
+            Integer.parseInt(test);
+        } catch(NumberFormatException e) {
```
Comment (KinTatHo) [inline src/main/java/seedu/address/model/person/Id.java:15]: syntax spacing i guess

File: src/main/java/seedu/address/model/util/SampleDataUtil.java [hunk 6]
```diff
@@ -6,12 +6,7 @@
 
 import seedu.address.model.AddressBook;
 import seedu.address.model.ReadOnlyAddressBook;
-import seedu.address.model.person.Address;
-import seedu.address.model.person.Email;
-import seedu.address.model.person.Name;
-import seedu.address.model.person.Person;
-import seedu.address.model.person.Phone;
-import seedu.address.model.person.Subject;
+import seedu.address.model.person.*;
```
Comment (KinTatHo) [inline src/main/java/seedu/address/model/util/SampleDataUtil.java:9]: import one by one i think safer

File: src/main/java/seedu/address/storage/JsonAdaptedPerson.java [hunk 7]
```diff
@@ -10,12 +10,7 @@
 import com.fasterxml.jackson.annotation.JsonProperty;
 
 import seedu.address.commons.exceptions.IllegalValueException;
-import seedu.address.model.person.Address;
-import seedu.address.model.person.Email;
-import seedu.address.model.person.Name;
-import seedu.address.model.person.Person;
-import seedu.address.model.person.Phone;
-import seedu.address.model.person.Subject;
+import seedu.address.model.person.*;
```
Comment (KinTatHo) [inline src/main/java/seedu/address/storage/JsonAdaptedPerson.java:13]: this too

**Review by KinTatHo — CHANGES_REQUESTED** [review]

File: src/main/java/seedu/address/logic/parser/AddressBookParser.java (hunk 3 above)
Comment (MarcusGitty) [reply src/main/java/seedu/address/logic/parser/AddressBookParser.java]: I think so too

File: src/main/java/seedu/address/logic/commands/ViewCommand.java (hunk 1 above)
Comment (MarcusGitty) [inline src/main/java/seedu/address/logic/commands/ViewCommand.java:23]: I'm Currently using ViewCommand.java for viewing of summary stats, hmm maybe i should change it to StatsCommand ?

Comment (0nesheep) [reply src/main/java/seedu/address/logic/commands/ViewCommand.java]: I will try combining with view first! But view is getting really long so in the future we can abstract it out

**Review by KinTatHo — APPROVED** [review]
LGTM

**Pull request merged**